src/Tizen.NUI/src/internal/Utility/ColorCutQuantizer.cs

```diff
@@ -0,0 +1,646 @@
+            {
+                T tmp = heap[i];
+
+                heap[i] = heap[j];
+                heap[j] = tmp;
+            }
+
+            private static int Parent(int i)
+            {
+                return (i + 1) / 2 - 1;
+            }
+
+            private static int YoungChild(int i)
+            {
+                return (i + 1) * 2 - 1;
+            }
+
+            private static int OldChild(int i)
+            {
+                return YoungChild(i) + 1;
+            }
+
+            private void Grow()
+            {
+                int newcapacity = capacity * growFactor + minGrow;
+                var newHeap = new T[newcapacity];
+
+                Array.Copy(heap, newHeap, capacity);
+                heap = newHeap;
+                capacity = newcapacity;
+            }
+
+            private bool Dominates(T x, T y)
+            {
+                return Comparer.Compare(x, y) <= 0;
+            }
+
+        }
+
+        /// <summary>
+        /// Comparator which sorts Vbox instances based on their volume, in descending order
+        /// </summary>
+        private sealed class VboxComparatorVolume : Comparer<Vbox>
+        {
+            public override int Compare(Vbox lhs, Vbox rhs)
+            {
+                return rhs.GetVolume() - lhs.GetVolume();
+            }
+        }
+
+        /// <summary>
+        /// Represents a tightly fitting box around a color space.
+        /// </summary>
+        private sealed class Vbox
+        {
+            private int lowerIndex;
+            private int upperIndex;
+            private int minRed, maxRed, minGreen, maxGreen, minBlue, maxBlue;
+
+            public Vbox(int lowerIndex, int upperIndex)
+            {
+                this.lowerIndex = lowerIndex;
+                this.upperIndex = upperIndex;
+                FitBox();
+            }
+
+            public int GetVolume()
+            {
+                return (maxRed - minRed + 1) * (maxGreen - minGreen + 1) * (maxBlue - minBlue + 1);
+            }
+
+            public Boolean CanSplit()
+            {
+                return GetColorCount() > 1;
+            }
+
+            public int GetColorCount()
+            {
+                return upperIndex - lowerIndex;
+            }
+
+            /// <summary>
+            /// Recomputes the boundaries of this box to tightly fit the colors within the box.
+            /// </summary>
+            public void FitBox()
+            {
+                // Reset the min and max to opposite values
+                minRed = minGreen = minBlue = 0xff;
+                maxRed = maxGreen = maxBlue = 0x0;
+                for (int i = lowerIndex; i <= upperIndex; i++)
+                {
+                    int red = (colors[i] >> 16) & 0xff;
+                    int green = (colors[i] >> 8) & 0xff;
+                    int blue = colors[i] & 0xff;
+                    if (red > maxRed)
+                    {
+                        maxRed = red;
```

Comment:
How about using trinomial operator?
maxRed = red > maxRed ? red : maxRed;

Reply:
That would be better.
Fix done.